framework/files/usr/lib/systemd/system/elemental-install.service

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Elemental Operator Registration
+Wants=network-online.target
+After=network-online.target
+
+[Install]
+WantedBy=multi-user.target
+
+[Service]
+Type=oneshot
+EnvironmentFile=-/etc/default/%N
+EnvironmentFile=-/etc/sysconfig/%N
+TimeoutStartSec=0
+StandardOutput=journal+console
+StandardError=journal+console
+ExecStartPre=/bin/sh -c 'echo "ELEMENTAL_TARGET=/dev/$(lsblk -r -o NAME,TYPE | grep -w disk | grep -v fd0 | awk \'{print $1}\')" > /etc/sysconfig/elemental-install'
```

Comment:
I'm still nervous about this line. I think the value set here is taking precedence over what set in the MachineRegistration

Reply:
My guess: to provide a default if no value is provided by the MachineRegistration. If this is the case, it could make sense because you can try to detect the device you want to install to only when you are on the actual host for the installation.
But the main point is: does the ELEMENTAL_TARGET takes precedence over the yaml config passed to "elemental install" ?

Reply:
We probably need validation then to make sure that at least a target is added OR make elemental-installer use an "auto" stanza and make it do the same as here instead of hardcoding files with something dangerous.


I can't recall, but IIRC env args have precedence over config files. I would need to recheck this thougth

Reply:
I completely agree with you. This really looks a bit hacky and is unexpected (or even buggy, if it takes precedence)

Reply:
This might be a moot point: https://github.com/rancher/elemental/pull/156#issuecomment-1175968783